src/main/java/com/pusher/client/channel/impl/ChannelImpl.java

```diff
@@ -213,4 +209,17 @@ private void validateArguments(final String eventName, final SubscriptionEventLi
                     "Cannot bind or unbind to events on a channel that has been unsubscribed. Call Pusher.subscribe() to resubscribe to this channel");
         }
     }
+
+    protected Set<SubscriptionEventListener> getInterestedListeners(String event) {
+        final Set<SubscriptionEventListener> listeners;
+        synchronized (lock) {
+            final Set<SubscriptionEventListener> sharedListeners = eventNameToListenerMap.get(event);
+            if (sharedListeners != null) {
+                listeners = new HashSet<SubscriptionEventListener>(sharedListeners);
+            } else {
+                listeners = null;
+            }
+        }
+        return listeners;
```

Comment:
Because there's nothing else in this method other than the synchronized block, we don't really need the local variable, you can just replace the assignments of `listeners` with `return` statements.

Reply:
Ah, that's very interesting. So Android Studio helped me simplify it even more. The if listeners = null is redundant, because if it's null we can just return... null - which then means we can simplify it down to just:

 ```
protected Set<SubscriptionEventListener> getInterestedListeners(String event) {
        synchronized (lock) {
            return eventNameToListenerMap.get(event);
        }
    }
```

Reply:
I don't think that's equivalent. The old code returned *a copy* of the collection of listeners, not the actual collection from the map.

Because Java's collections are all mutable by default, it's important to copy the collection returned, so that if the caller manipulates it they aren't making unsynchronized changes to the contents of the `eventNameToListenerMap`.

Reply:
Hmm. Good to know! I've updated this code to return a copy of the interested listeners - it's still a little bit simplified than what was there before.